### PR TITLE
ci: replace black with ruff format

### PR DESCRIPTION
**Pull request by weirdtangent**
ruff already lints this repo and its formatter is black-compatible, so black was a second tool, a second pinned version and a second dev dependency for no added coverage.

**Changes**
- CI step `Black format check` → `Ruff format check` (`ruff format --check .`)
- pre-commit hook `black` → `ruff-format` (`uv run ruff format .`)
- black removed from both dependency lists, `[tool.black]` deleted, `uv.lock` regenerated so `uv sync` stops installing it
- **CI ruff pin aligned to pyproject's 0.15.7** (CI pinned `0.15.0` while pyproject pinned `0.15.7`; black was likewise skewed `26.1.0` vs `26.3.1`). That skew barely mattered for linting but matters a lot now that CI *formats* — a version difference would fail CI in a way nobody could reproduce locally. Verified 0.15.0 and 0.15.7 produce byte-identical formatting output here before aligning.

`[tool.ruff.format]` was already configured in this repo, so `skip-magic-trailing-comma` and `docstring-code-format` now actually apply.

**Reformats 4 files** — joins implicitly concatenated f-strings and explodes two long call sites. In `tests/test_llm.py` that moves a `# type: ignore[arg-type]` onto the argument line; since the pre-commit mypy hook runs `mypy --ignore-missing-imports .` and *does* cover `tests/`, I verified it reports **zero errors both before and after**.

The job name is unchanged and `claude` is the only required status check here, so branch protection is unaffected.

`ruff check`, `ruff format --check` and all **1929 tests** pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Tooling and formatting-only changes with no runtime logic edits; lockfile bumps are minor dependency updates.
> 
> **Overview**
> **Replaces Black with Ruff as the sole Python formatter** across CI, pre-commit, and contributor docs, and drops Black from `pyproject.toml`, dev dependencies, and `uv.lock`.
> 
> CI now runs **`ruff format --check .`** instead of `black --check`, installs **ruff 0.15.7** (aligned with the repo pin; Black was removed from the install line), and pre-commit swaps the **`black`** hook for **`ruff-format`**. Existing **`[tool.ruff.format]`** settings in pyproject now govern formatting end-to-end.
> 
> The diff includes **mechanical reformatting** in `bin/kiosk-mqtt-listener.py`, `pulse/assistant/llm.py`, `pulse/overlay.py`, and `tests/test_llm.py` (e.g. joined f-strings, wrapped long calls). **`uv.lock`** also reflects dependency cleanup (Black removed) and routine bumps such as **websockets 17.0.1** and **icalendar 7.2.2**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit cec8058b579415f299cb0639d9dcfee862fc6e42. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -62,7 +57,6 @@ docstring-code-format = true
 
 [dependency-groups]
 dev = [
-    "black>=24.8.0",
     "mypy>=2.3.0",
     "pip-audit>=2.10.1",
     "pre-commit>=4.6.1",
```
[the remark this answers is not in the document]
Comment (weirdtangent) [reply pyproject.toml:62]: Correct, and fixed in cec8058 — `[dependency-groups] dev` now pins `ruff==0.15.7` to match `[project.optional-dependencies] dev` and CI.

This is the third instance of the same class of problem in this batch, which is a good argument that it is worth being strict about: a lint-only version spread is cosmetic, but a *formatter* version spread produces green-locally / red-in-CI with nothing the contributor can do to reproduce it. The other two were blink2mqtt's `ruff>=0.16.2` vs `>=0.14.1` split, and this repo's CI pinning 0.15.0 against pyproject's 0.15.7 (already addressed earlier in this PR).

Verified `uv run ruff --version` now reports 0.15.7, and both `ruff check` and `ruff format --check` pass.
